x-pack/plugins/security_solution/scripts/endpoint/common/fleet_services.ts

```diff
@@ -209,9 +215,70 @@ export const waitForHostToEnroll = async (
     );
   }
 
-  log.debug(`Host [${hostname}] has been enrolled with fleet`);
+  log.info(`Host [${hostname}] has been enrolled with fleet`);
   log.verbose(found);
 
+  if (!awaitMetadata) {
+    return found;
+  }
+  log.info('Querying indices');
+
+  log.debug(
+    JSON.stringify(
+      await esClient?.search({
```

Comment:
This is the correct place to do this check (against endpoint metadata indexes).
This module - `fleet_services` - implies a focus on Fleet functionality. The method - `waitForHostToEnroll()` - as the description indicates, just waits for the host to show up in the Fleet's Agent list. That will fulfill the responsibility of this function.  This change also assumes that the host being enrolled is one running with the Endpoint integrations - which is a bad assumption.

I also don't think you need this - we already have `waitForEndpointToStreamData()` in the `endpoint_metadata_services` that essentially does this: waits for an endpoint to "show up" on the Endpoint list - see here: https://github.com/elastic/kibana/blob/797104b4d010ed7543b72ca679dcb762d7e0e62a/x-pack/plugins/security_solution/scripts/endpoint/common/endpoint_metadata_services.ts#L153

Reply:
`waitForEndpointToStreamData()` does `fetchEndpointMetadata()`, which
queries a specific endpoint, while what I did here queries the list. I'm
aware that we already do `waitForEndpointToStreamData`, but it didn't
prevent the endpoint from not coming up in Kibana where we query the
metadata list. I added it because I wasn't sure if there was any other
purpose of `waitForEndpointToStreamData`, so I wonder, how about
switching the query to the list and verifying that the host is on the
metadata list? It seems to have solved the problem, even though we might
not yet know why ;)

Reply:
Hmmm. But why would queuing the Endpoint list provide different results than fetching the metadata for a single endpoint? They both use the same data source (the transform unified index)

That being said - doing queries agains Endpoint indexes from here is not the correct approach since not all agents might be running endpoint integration.

Reply:
Yup, Im aware that the source for both these operations is the same, however, the outcome varied. Tests were failing on checking endpoint page for enrolled endpoints, this happens after the setup task that queries metadata for a single host. Could you suggest a better place for such a check?